### PR TITLE
chore(github) modify pull request template

### PR DESCRIPTION
**Pull request by kikito**
### Summary

Modify the pull request template with a checklist so that we don't forget
commonly needed items:

* tests
* changelog
* docs

### CODE REVIEW

**Review by RobSerafini — CHANGES_REQUESTED** [review]

**Review by dndx — APPROVED** [review]

**Comment by dndx** [comment]
@hbagdi is this ready to merge? If so, please remove the `do not merge` label.

**Review by hbagdi — APPROVED** [review]

**Pull request merged**
